### PR TITLE
user servise with bag

### PR DESCRIPTION
**Pull request by Vladimir2430**

### CODE REVIEW

File: src/app/models/user.model.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export class User {
+  $key: string;
```
Comment (FrontendMentor) [inline src/app/models/user.model.ts]: Please, remove extra space

File: src/app/servise/user.servise.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { Observable } from 'rxjs/Observable';
+import { AngularFireDatabase, AngularFireList } from 'angularfire2/database';
+import { User } from '../models/user.model';
+import * as firebase from 'firebase/app';
+import { Router } from '@angular/router';
+
+@Injectable()
+export class UserService {
+	userList: AngularFireList<any>;
+ 	selectedUser: User = new User();
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: same here

File: src/app/servise/user.servise.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  cotnstructor(private firebase: AngularFireDatabase) {  }
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: It seems like there is a misprint. It should read `constructor`. And add new line before.

File: src/app/servise/user.servise.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  updateUser(user: User) {
+    this.userList.update(user.$key,{
+      id: user.id,
+      name: user.name
+    });
+  }
+
+  deleteUser($key: string) {
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: Could you set return type for all methods?

File: src/app/models/user.model.ts (hunk 1 above)
Comment (Vladimir2430) [reply src/app/models/user.model.ts]: ok

File: src/app/servise/user.servise.ts (hunk 2 above)
Comment (Vladimir2430) [reply src/app/servise/user.servise.ts]: ok

File: src/app/servise/user.servise.ts (hunk 3 above)
Comment (Vladimir2430) [reply src/app/servise/user.servise.ts]: thank you )))

File: src/app/servise/user.servise.ts (hunk 4 above)
Comment (Vladimir2430) [reply src/app/servise/user.servise.ts]: ok

File: src/app/servise/user.servise.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { AngularFireDatabase, AngularFireList } from 'angularfire2/database';
+import { User } from '../models/user.model';
+import * as firebase from 'firebase/app';
+import { Router } from '@angular/router';
+
+@Injectable()
+export class UserService {
+	userList: AngularFireList<any>;
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: please check the space before variable

File: src/app/servise/user.servise.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+ 	selectedUser: User = new User();
+    constructor(private firebase: AngularFireDatabase) {}
+
+    getUser() {
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: Could you add return type for this method? For example `getUser(): any` if you don't know what should return `this.userList`, or you can try set `getUser(): User[]`

File: src/app/servise/user.servise.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    	this.userList = this.firebase.list('users');
+    	return this.userList;
+    }
+
+    insertUser(user: User) {
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: This method should return the type `void`. Please change it to `insertUser(user: User): void {`

File: src/app/servise/user.servise.ts [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+        	name: user.name
+        });
+    }
+
+    updateUser(user: User) {
+        this.userList.update(user.$key,{
+        	id: user.id,
+        	name: user.name
+        });
+    }
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: same here

File: src/app/servise/user.servise.ts [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+    deleteUser($key: string) {
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: same here

File: src/app/servise/user.servise.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+		this.userList.remove($key);
+		this.userList.remove(name);
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: please check the space before `this.userList.remove($key); ` and `this.userList.remove(name);`

File: src/app/users/user.component.ts [hunk 11]
```diff
@@ -1,11 +1,13 @@
...
+    selector: 'app-user',
+    templateUrl: './user.component.html',
+    styleUrls: ['./user.component.css']
 })
-export class UserComponent {
-  users: User[];
+export class UserComponent implements OnInit {
+    constructor() {}
+
+    ngOnInit() {}
```
Comment (FrontendMentor) [inline src/app/users/user.component.ts]: remove ngOnInit and constructor if you are not using it at the moment

File: src/index.html [hunk 12]
```diff
@@ -1,14 +1,16 @@
...
 
-  <meta name="viewport" content="width=device-width, initial-scale=1">
-  <link rel="icon" type="image/x-icon" href="favicon.ico">
-</head>
-<body>
-  <app-root></app-root>
-</body>
+		<meta name="viewport" content="width=device-width, initial-scale=1">
+    	<link rel="icon" type="image/x-icon" href="favicon.ico">
+    	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/css/bootstrap.min.css">
```
Comment (FrontendMentor) [inline src/index.html]: I'm not sure that you should import the bootstrap styles, I've added them previously (https://github.com/incubator-front/FAQ-Team1/commit/976c10dfbd4d2d16d57c494b413e68383495facb). Could you check it?

File: src/app/servise/user.servise.ts (hunk 5 above)
Comment (Vladimir2430) [reply src/app/servise/user.servise.ts]: ok

File: src/app/servise/user.servise.ts (hunk 7 above)
Comment (Vladimir2430) [reply src/app/servise/user.servise.ts]: ok

File: src/app/servise/user.servise.ts (hunk 8 above)
Comment (Vladimir2430) [reply src/app/servise/user.servise.ts]: ok

File: src/app/servise/user.servise.ts (hunk 10 above)
Comment (Vladimir2430) [reply src/app/servise/user.servise.ts]: ok

File: src/app/users/user.component.ts (hunk 11 above)
Comment (Vladimir2430) [reply src/app/users/user.component.ts]: ok

File: src/app/servise/user.servise.ts [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+import * as firebase from 'firebase/app';
+import { Router } from '@angular/router';
+
+@Injectable()
+export class UserService {
+	userList: AngularFireList <any>;
+ 	selectedUser: User = new User();
+    constructor(private firebase: AngularFireDatabase) {}
+
+    getUser() {
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: This method should return `AngularFireList<any>`

File: src/app/servise/user.servise.ts [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+    	return this.userList = this.firebase.list('users') ? this.userList : this.firebase.list('users');
+    }
+
+    insertUser(user: User): void {
+    this.userList.push({
```
Comment (FrontendMentor) [inline src/app/servise/user.servise.ts]: fix space here

**Pull request merged**